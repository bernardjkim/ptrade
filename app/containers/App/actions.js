/*
 *
 * App actions
 *
 */

import {
  LOAD_TOKEN,
  CREATE_SESSION,
  CREATE_SESSION_ERROR,
  CREATE_SESSION_SUCCESS,
  DELETE_SESSION,
} from './constants';

/**
 * Load current jwt token if available
 *
 * @return {object} An action object with a type of LOAD_TOKEN
 *
 */
export function loadToken() {
  const token = sessionStorage.getItem('jwtToken');
  return {
    type: LOAD_TOKEN,
    token,
  };
}

/**
 * Delete the current session token
 *
 * @return {object} An action object with a type of DELETE_TOKEN
 *
 */
export function deleteSession() {
  sessionStorage.removeItem('jwtToken');
  return {
    type: DELETE_SESSION,
  };
}

/**
 * Create session, this action starts the request saga
 *
 * @return {object} An action object with a type of CREATE_SESSION
 */
export function createSession() {
  return {
    type: CREATE_SESSION,
  };
}

/**
 * Dispatched when the session token is loaded by the request saga
 *
 * @param  {string} token The chart data
 *
 * @return {object}       An action object with a type of CREATE_SESSION_SUCCESS
 *                        passing the token
 */
export function createSessionSuccess(token) {
  return {
    type: CREATE_SESSION_SUCCESS,
    token,
  };
}

/**
 * Dispatched when createing the session fails
 *
 * @param  {object} error The error
 *
 * @return {object}       An action object with a type of CREATE_SESSION_ERROR
 *                        passing the error
 */
export function createSessionError(error) {
  return {
    type: CREATE_SESSION_ERROR,
    error,
  };
}
