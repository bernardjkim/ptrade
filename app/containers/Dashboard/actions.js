/*
 *
 * Dashboard actions
 *
 */

import {
  CHANGE_SEARCH,
  CHANGE_TIME_FRAME,
  SELECT_SYMBOL,
  LOAD_CHART,
  LOAD_CHART_SUCCESS,
  LOAD_CHART_ERROR,
  LOAD_QUOTE,
  LOAD_QUOTE_SUCCESS,
  LOAD_QUOTE_ERROR,
} from './constants';

/**
 * Change the current search symbol
 *
 * @param  {string} search  A stock symbol
 *
 * @return {object}         An action object with a type of CHANGE_SEARCH
 */
export function changeSearch(search) {
  return {
    type: CHANGE_SEARCH,
    search,
  };
}

/**
 * Change the current selected symbol
 *
 * @param  {string} symbol  Stock symbol
 *
 * @return {object}         An action object with a type of SELECT_SYMBOL
 */
export function selectSymbol(symbol) {
  return {
    type: SELECT_SYMBOL,
    symbol,
  };
}

/**
 * Change the selected time frame
 *
 * @param  {number} tf  Time frame index
 *
 * @return {object}     An action object with a type of CHANGE_TIME_FRAME
 */
export function changeTimeFrame(tf) {
  return {
    type: CHANGE_TIME_FRAME,
    tf,
  };
}

/**
 * Load chart, this action starts the request saga
 *
 * @return {object} An action object with a type of LOAD_CHART
 */
export function loadChart() {
  return {
    type: LOAD_CHART,
  };
}

/**
 * Dispatched when the chart is loaded by the request saga
 *
 * @param  {array} chart  The chart data
 *
 * @return {object}       An action object with a type of LOAD_CHART_SUCCESS passing the data
 */
export function chartLoaded(chart) {
  return {
    type: LOAD_CHART_SUCCESS,
    chart,
  };
}

/**
 * Dispatched when loading the chart fails
 *
 * @param  {object} error The error
 *
 * @return {object}       An action object with a type of LOAD_CHART_ERROR passing the error
 */
export function chartLoadingError(error) {
  return {
    type: LOAD_CHART_ERROR,
    error,
  };
}

/**
 * Load the stock quote, this action starts the request saga
 *
 * @return {object} An action object with a type of LOAD_QUOTE
 */
export function loadQuote() {
  return {
    type: LOAD_QUOTE,
  };
}

/**
 * Dispatched when the stock quote is loaded by the request saga
 *
 * @param  {object} quote The stock quote
 *
 * @return {object}       An action object with a type of LOAD_QUOTE_SUCCESS passing the quote
 */
export function quoteLoaded(quote) {
  return {
    type: LOAD_QUOTE_SUCCESS,
    quote,
  };
}

/**
 * Dispatched when loading the stock quote fails
 *
 * @param  {object} error The error
 *
 * @return {object}       An action object with a type of LOAD_QUOTE_ERROR passing the error
 */
export function quoteLoadingError(error) {
  return {
    type: LOAD_QUOTE_ERROR,
    error,
  };
}
