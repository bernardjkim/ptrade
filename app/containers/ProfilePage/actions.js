/*
 *
 * ProfilePage actions
 *
 */

import {
  CHANGE_TIME_FRAME,
  CHANGE_TRANSFER_AMOUNT,
  LOAD_BALANCE,
  LOAD_BALANCE_ERROR,
  LOAD_BALANCE_SUCCESS,
  LOAD_POSITIONS,
  LOAD_POSITIONS_ERROR,
  LOAD_POSITIONS_SUCCESS,
  LOAD_CHART,
  LOAD_CHART_ERROR,
  LOAD_CHART_SUCCESS,
  REQUEST_TRANSFER,
  REQUEST_TRANSFER_ERROR,
  REQUEST_TRANSFER_SUCCESS,
} from './constants';

export function changeTimeFrame(timeFrame) {
  return {
    type: CHANGE_TIME_FRAME,
    timeFrame,
  };
}

export function changeTransferAmount(amount) {
  return {
    type: CHANGE_TRANSFER_AMOUNT,
    amount,
  };
}

export function loadBalance() {
  return {
    type: LOAD_BALANCE,
  };
}

export function loadBalanceSuccess(balance) {
  return {
    type: LOAD_BALANCE_SUCCESS,
    balance,
  };
}

export function loadBalanceError(error) {
  return {
    type: LOAD_BALANCE_ERROR,
    error,
  };
}

export function loadPositions() {
  return {
    type: LOAD_POSITIONS,
  };
}

export function loadPositionsSuccess(positions) {
  let total = 0;
  positions.forEach(p => {
    total += p.shares * p.price_per_share;
  });
  return {
    type: LOAD_POSITIONS_SUCCESS,
    positions,
    total,
  };
}

export function loadPositionsError(error) {
  return {
    type: LOAD_POSITIONS_ERROR,
    error,
  };
}

export function loadChart() {
  return {
    type: LOAD_CHART,
  };
}

export function loadChartSuccess(chart) {
  return {
    type: LOAD_CHART_SUCCESS,
    chart,
  };
}

export function loadChartError(error) {
  return {
    type: LOAD_CHART_ERROR,
    error,
  };
}

export function requestTransfer() {
  return {
    type: REQUEST_TRANSFER,
  };
}

export function requestTransferSuccess() {
  return {
    type: REQUEST_TRANSFER_SUCCESS,
  };
}

export function requestTransferError(error) {
  return {
    type: REQUEST_TRANSFER_ERROR,
    error,
  };
}
