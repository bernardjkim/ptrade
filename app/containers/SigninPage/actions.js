/*
 *
 * SigninPage actions
 *
 */

import { CHANGE_INPUT } from './constants';

/**
 * Change the input value for specified field
 *
 * @param  {string} field The field to be changed
 * @param  {string} value The value to be set
 *
 * @return {object}       An object of type CHANGE_INPUT
 */
export function changeInput(field, value) {
  return {
    type: CHANGE_INPUT,
    field,
    value,
  };
}
