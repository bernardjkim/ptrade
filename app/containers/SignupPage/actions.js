/*
 *
 * SignupPage actions
 *
 */

import {
  CHANGE_INPUT,
  CREATE_USER,
  CREATE_USER_ERROR,
  CREATE_USER_SUCCESS,
} from './constants';

/**
 * Change the input value for specified field
 *
 * @param  {string} field The field to be changed
 * @param  {string} value The value to be set
 *
 * @return {object}       An object of type CHANGE_INPUT
 */
export function changeInput(field, value) {
  return {
    type: CHANGE_INPUT,
    field,
    value,
  };
}

/**
 * Create user, this action starts the request saga
 *
 * @return {object} An action object with a type of CREATE_USER
 */
export function createUser() {
  return {
    type: CREATE_USER,
  };
}

/**
 * Dispatched when the user is successfuly created
 *
 * @return {object} An action object with a type of CREATE_USER_SUCCESS
 *
 */
export function createUserSuccess() {
  return {
    type: CREATE_USER_SUCCESS,
  };
}

/**
 * Dispatched when creating the user fails
 *
 * @param  {object} error The error
 *
 * @return {object}       An action object with a type of CREATE_USER_ERROR
 *                        passing the error
 */
export function createUserError(error) {
  return {
    type: CREATE_USER_ERROR,
    error,
  };
}
