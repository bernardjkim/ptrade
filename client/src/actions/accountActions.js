import axios from 'axios';
import qs from 'qs';

import * as auth from '../system/auth';

export function check() {
    const authRequest = {
        method: 'GET',
        url: 'http://localhost:8080/auth/check',
        headers: { 'X-App-Token': auth.getCookie('api.example.com') },
    }

    return function (dispatch) {
        dispatch({
            type: 'ACCOUNT_CHECK',
            payload: axios(authRequest),
        })
            .catch(error => {
                console.log(error);
            });
    }
}

export function create(user) {
    const { first, last, email, password } = user;
    const authRequest = {
        method: 'POST',
        url: 'http://localhost:8080/auth/signup',
        data: qs.stringify({
            firstName: first,
            lastName: last,
            email: email,
            password: password,
        }),
    }

    return function (dispatch) {
        dispatch({
            type: 'ACCOUNT_CREATE',
            payload: axios(authRequest),
        })
            .then((response) => {
                dispatch(signIn({ email, password }));

            })
            .catch(error => {
                console.log(error);
            });
    }
}

export function signIn(user) {
    const authRequest = {
        method: 'POST',
        url: 'http://localhost:8080/auth/login',
        data: qs.stringify({ email: user.email, password: user.password }),
    }

    return function (dispatch) {
        dispatch({
            type: 'ACCOUNT_SIGNIN',
            payload: axios(authRequest),
        })
        .then((response) => {
            dispatch(check());
        })
            .catch(error => {
                console.log(error);
            });
    }
}

export function signOut() {
    return {
        type: 'ACCOUNT_SIGNOUT',
    }
}