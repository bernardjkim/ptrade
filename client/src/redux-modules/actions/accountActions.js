import axios from 'axios';
import qs from 'qs';

import * as auth from '../../system/auth';
import { account } from './types';


export function check() {
    const authRequest = {
        method: 'GET',
        url: 'http://localhost:8080/auth/check',
        headers: { 'X-App-Token': auth.getCookie('api.example.com') },
    }

    return function (dispatch) {
        dispatch({
            type: account.CHECK,
            payload: axios(authRequest),
        })
            .catch(error => {
                console.log(error);
            });
    }
}

export function create(user) {
    const { first, last, email, password } = user;
    const authRequest = {
        method: 'POST',
        url: 'http://localhost:8080/auth/signup',
        data: qs.stringify({
            firstName: first,
            lastName: last,
            email: email,
            password: password,
        }),
    }

    return function (dispatch) {
        dispatch({
            type: account.CREATE,
            payload: axios(authRequest),
        })
            .then((response) => {
                dispatch(signIn({ email, password }));

            })
            .catch(error => {
                console.log(error);
            });
    }
}

export function signIn(user) {
    const authRequest = {
        method: 'POST',
        url: 'http://localhost:8080/auth/login',
        data: qs.stringify({ email: user.email, password: user.password }),
    }

    return function (dispatch) {
        dispatch({
            type: account.SIGNIN,
            payload: axios(authRequest),
        })
        .then((response) => {
            dispatch(check());
        })
            .catch(error => {
                console.log(error);
            });
    }
}

export function signOut() {
    return {
        type: account.SIGNOUT,
    }
}
