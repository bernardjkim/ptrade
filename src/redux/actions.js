import axios from 'axios';
import qs from 'qs';

import * as auth from 'system/auth';

import {
    SESSION_CREATE_START,
    SESSION_CREATE_SUCCESS,
    SESSION_CREATE_FAIL,
    USER_CREATE_START,
    USER_CREATE_SUCCESS,
    USER_CREATE_FAIL,
    VALIDATE_START,
    VALIDATE_SUCCESS,
    VALIDATE_FAIL,
} from './types';

// Send a request to the ptrade api to create a new session with the
// given email and password.
export const signin = (email, password) => {
    const createSessionRequest = {
        method: 'POST',
        url: process.env.API_URL + '/sessions',
        data: qs.stringify({ email: email, password: password }),
    };
    return (dispatch) => {
        dispatch({ type: SESSION_CREATE_START })
        axios(createSessionRequest)
            .then(response => {
                dispatch({ type: SESSION_CREATE_SUCCESS, payload: response.data, });
                dispatch(validate());
            })
            .catch(error => {
                dispatch({ type: SESSION_CREATE_FAIL, payload: error, })
            });
    };
}

// Send a request to the ptrade api to create a new user with the
// given email and password.
export const signup = (first, last, email, password) => {
    const createUserRequest = {
        method: 'POST',
        url: process.env.API_URL + '/users',
        data: qs.stringify({
            first: first,
            last: last,
            email: email,
            password: password,
        }),
    }
    return (dispatch) => {
        dispatch({ type: USER_CREATE_START })
        axios(createUserRequest)
            .then(response => {
                dispatch({ type: USER_CREATE_SUCCESS, payload: response.data, });
                dispatch(signin(email, password));
            })
            .catch(error => {
                dispatch({ type: USER_CREATE_FAIL, payload: error, })
            });
    };
}

// Send a request to ptrade api to validate the current Session-Token.
export const validate = () => {
    const validateRequest = {
        method: 'GET',
        url: process.env.API_URL + '/sessions/validate',
        headers: { 'Session-Token': auth.getCookie('api.ptrade.com') },
    }
    return (dispatch) => {
        dispatch({ type: VALIDATE_START })
        axios(validateRequest)
            .then(response => {
                dispatch({ type: VALIDATE_SUCCESS, payload: response.data, });
            })
            .catch(error => {
                dispatch({ type: VALIDATE_FAIL, payload: error, })
            });
    };
}