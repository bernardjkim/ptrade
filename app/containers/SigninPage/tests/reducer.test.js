import { fromJS } from 'immutable';
import signinPageReducer from '../reducer';
import { changeInput } from '../actions';

describe('signinPageReducer', () => {
  let state;
  beforeEach(() => {
    state = fromJS({
      email: false,
      password: false,
      loading: false,
      error: false,
    });
  });
  it('returns the initial state', () => {
    expect(signinPageReducer(undefined, {})).toMatchSnapshot();
  });

  it('handles the changeInput action', () => {
    expect(
      signinPageReducer(state, changeInput('email', 'test@email.com')),
    ).toMatchSnapshot();
  });
});
