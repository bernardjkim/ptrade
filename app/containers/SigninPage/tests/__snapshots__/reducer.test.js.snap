// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`signinPageReducer handles the changeInput action 1`] = `
Immutable.Map {
  "email": "test@email.com",
  "password": false,
  "loading": false,
  "error": false,
}
`;

exports[`signinPageReducer returns the initial state 1`] = `
Immutable.Map {
  "email": false,
  "password": false,
  "loading": false,
  "error": false,
}
`;
